airbyte-commons/src/main/java/io/airbyte/commons/json/JsonSecretsProcessor.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.commons.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Preconditions;
+
+public class JsonSecretsProcessor {
+
+  public static String AIRBYTE_SECRET_FIELD = "airbyte_secret";
+  private static String PROPERTIES_FIELD = "properties";
+
+  /**
+   * Returns a copy of the input object wherein any fields annotated with "airbyte_secret" in the
+   * input schema are removed.
+   * <p>
+   * TODO this method only removes secrets at the top level of the configuration object. It does not support the keywords anyOf, allOf, oneOf, not, and
+   * dependencies. This will be fixed in the future.
+   *
+   * @param schema Schema containing secret annotations
+   * @param obj    Object containing potentially secret fields
+   * @return
+   */
+  public JsonNode removeSecrets(JsonNode obj, JsonNode schema) {
+    assertValidSchema(schema);
+    Preconditions.checkArgument(schema.isObject());
+
+    ObjectNode properties = (ObjectNode) schema.get(PROPERTIES_FIELD);
+    JsonNode copy = obj.deepCopy();
```

Comment:
`Jsons.clone` is an option too. I don't know what guarantees `deepCopy` gives but if you do, then good to go!

Reply:
deepCopy can return a different type e.g: `ObjectNode`. Just a small thing. I didn't realize jsons.clone was a thing